tgui/packages/tgui/interfaces/ClockworkSlab.js

```diff
@@ -0,0 +1,322 @@
+          </b>
+          A powerful tool that can rapidly construct Brass structures, or
+          convert most materials to Brass.
+          <br />
+          <b>
+            <font color="#DED09F">Cogscarab:&nbsp;</font>
+          </b>
+          A small drone possessed by the spirits of the fallen soldiers which
+          will protect Reebe while you go out and spread the truth!
+          <br />
+          <b>
+            <font color="#FF9D9D">Clockwork Marauder:&nbsp;</font>
+          </b>
+          A powerful shell that can deflect ranged attacks and delivers a strong
+          blow in close quarter combat.
+          <br />
+          <br />
+        </Section>
+      </Collapsible>
+    </Fragment>
+  );
+};
+
+export const ClockworkSpellList = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { selectedTab } = props;
+  const { scriptures = [] } = data;
+  return (
+    <Table>
+      {scriptures.map((script) =>
+        script.type === selectedTab ? (
+          <Fragment key={script}>
+            <TableRow>
+              <Table.Cell bold>{script.name}</Table.Cell>
+              <Table.Cell collapsing textAlign="right">
+                <Button
+                  fluid
+                  color={script.purchased ? 'default' : 'average'}
+                  content={
+                    script.purchased
+                      ? 'Invoke ' + convertPower(script.cost)
+                      : script.cog_cost + ' Cogs'
+                  }
+                  tooltip={script.tip}
+                  disabled={false}
+                  onClick={() =>
+                    act('invoke', {
+                      scriptureName: script.name,
+                    })
+                  }
+                />
+              </Table.Cell>
+            </TableRow>
+            <TableRow>
+              <Table.Cell>{script.desc}</Table.Cell>
+              <Table.Cell collapsing textAlign="right">
+                <Button
+                  fluid
+                  content={'Quickbind'}
+                  disabled={!script.purchased}
+                  onClick={() =>
+                    act('quickbind', {
+                      scriptureName: script.name,
+                    })
+                  }
+                />
+              </Table.Cell>
+            </TableRow>
+            <Table.Cell>
+              <Divider />
+            </Table.Cell>
+          </Fragment>
+        ) : (
+          <Box key={script} />
+        )
+      )}
+    </Table>
+  );
+};
+
+export const ClockworkOverview = (props, context) => {
+  const { data } = useBackend(context);
+  const { power, cogs, vitality } = data;
+  return (
+    <Box>
+      <Box color="good" bold fontSize="16px">
+        {'Celestial Gateway Report'}
+      </Box>
+      <Divider />
+      <ClockworkOverviewStat
+        title="Cogs"
+        amount={cogs}
+        maxAmount={cogs + 50 / cogs}
```

Comment:
This applies to the 2 stats below this as well, but all 3 of these vars can be 0 and thus should be `/ Math.max(cogs, 1)` (or something similar) or this code will crash if currently 0.

Also this maxAmount math is basically the same as 50 or 50,000 but produces an impression that the stats increase pseudo-logarithmically. Not sure if intentional.

Reply:
![image](https://user-images.githubusercontent.com/41448081/198892278-a167fabb-f30a-472b-8e03-e1cf50c308e1.png)

Seems to work fine